test/src/test-cppapi-consolidation-plan.cc

```diff
@@ -213,6 +215,225 @@ TEST_CASE_METHOD(
   Array array{ctx_, SPARSE_ARRAY_NAME, TILEDB_READ};
   ConsolidationPlan consolidation_plan(ctx_, array, 1024 * 1024);
 
-  // Check dump
-  CHECK(consolidation_plan.dump() == "Not implemented\n");
+  // Check dump.
+  CHECK(consolidation_plan.dump() == "{\n  \"nodes\": [\n  ]\n}\n");
+}
+
+TEST_CASE_METHOD(
+    CppConsolidationPlanFx,
+    "C++ API: Consolidation plan, de-interleave 1",
+    "[cppapi][consolidation_plan][de-interleave-1]") {
+  create_sparse_array();
+
+  // Write one fragment with NED [1, 4][1, 4] and one with NED [2, 6][2, 6].
+  // Since they intersect, they should be batched together.
+  auto uri1 = write_sparse({0, 1}, {1, 4}, {1, 4}, 1);
+  auto uri2 = write_sparse({2, 3}, {2, 6}, {2, 6}, 2);
+
+  // Create a consolidation plan with max fragment size of 1. That way, the
+  // two fragments above don't risk getting batched together when combining
+  // small fragments.
+  Array array{ctx_, SPARSE_ARRAY_NAME, TILEDB_READ};
+  ConsolidationPlan consolidation_plan(ctx_, array, 1);
+
+  // Validate the plan.
+  CHECK(consolidation_plan.num_nodes() == 1);
+  CHECK(consolidation_plan.num_fragments(0) == 2);
+  CHECK(consolidation_plan.fragment_uri(0, 0) == uri1);
+  CHECK(consolidation_plan.fragment_uri(0, 1) == uri2);
+}
+
+TEST_CASE_METHOD(
+    CppConsolidationPlanFx,
+    "C++ API: Consolidation plan, de-interleave 2",
+    "[cppapi][consolidation_plan][de-interleave-2]") {
+  create_sparse_array();
+
+  // Write one fragment with NED [1, 4][1, 4] and one with NED [2, 6][2, 6].
+  // Since they intersect, they should be batched together.
+  auto uri1 = write_sparse({0, 1}, {1, 4}, {1, 4}, 1);
+  auto uri2 = write_sparse({2, 3}, {2, 6}, {2, 6}, 2);
+
+  // Write one fragment with NED [10, 14][10, 14] and one with NED [12, 16][12,
+  // 16]. Since they intersect, they should be batched together.
+  auto uri3 = write_sparse({4, 5}, {10, 14}, {10, 14}, 3);
+  auto uri4 = write_sparse({6, 7}, {12, 16}, {12, 16}, 4);
+
+  // Create a consolidation plan with max fragment size of 1. That way, the
+  // two fragments above don't risk getting batched together when combining
+  // small fragments.
+  Array array{ctx_, SPARSE_ARRAY_NAME, TILEDB_READ};
+  ConsolidationPlan consolidation_plan(ctx_, array, 1);
+
+  // Validate the plan.
+  CHECK(consolidation_plan.num_nodes() == 2);
+  CHECK(consolidation_plan.num_fragments(0) == 2);
+  CHECK(consolidation_plan.fragment_uri(0, 0) == uri1);
+  CHECK(consolidation_plan.fragment_uri(0, 1) == uri2);
+  CHECK(consolidation_plan.fragment_uri(1, 0) == uri3);
```

Comment:
I am a bit scared of those checks because the order of nodes 0 and 1 in `fragment_uris_` comes from the order in `plan` which comes from the call to `partition` that according to spec"Relative order of the elements is not preserved" . So if I get that right, we might get random test failures just because of the ordering. Might be too simple of a case and produce always the same order, but thought of bringing this up to keep in mind for this and future tests.